app/Models/ImageMention.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Database\Eloquent\Model;
+use App\Models\Concerns\SortableTrait;
+use Illuminate\Support\Arr;
+
+/**
+ * Class ImageMention
+ * @package App\Models
+ *
+ * @property integer $image_id
+ * @property integer $post_id
+ * @property integer $entity_id
+ * @property Entity $entity
+ * @property Post $post
+ * @property Entity $target
+ *
+ * @method static self|Builder prepareCount()
+ */
+class ImageMention extends Model
+{
+    use SortableTrait;
+
+    public $fillable = [
+        'entity_id',
+        'image_id',
+        'post_id',
+    ];
+
+    protected $sortable = [
+        'name',
+        'type',
+    ];
+
+    /**
+     * @return \Illuminate\Database\Eloquent\Relations\BelongsTo
+     */
+    public function target()
+    {
+        return $this->belongsTo('App\Models\Images', 'image_id', 'id');
+    }
+
+    /**
+     * @return \Illuminate\Database\Eloquent\Relations\BelongsTo
+     */
+    public function entity()
+    {
+        return $this->belongsTo('App\Models\Entity', 'entity_id', 'id');
+    }
+
+    /**
+     * @return \Illuminate\Database\Eloquent\Relations\BelongsTo
+     */
+    public function post()
+    {
+        return $this->belongsTo('App\Models\Post', 'post_id', 'id');
+    }
+    
+    /**
+     * Determine if the mention goes to a post
+     * @return bool
+     */
+    public function isPost(): bool
+    {
+        return !empty($this->post_id);
+    }
+
+    /**
+     * Determine if the mention goes to an entity
+     * @return bool
+     */
+    public function isEntity(): bool
+    {
+        return !empty($this->entity_id);
+    }
+
+    /**
+     * Build the query that will loop on the various mentions to get the total count.
+     * The AclTrait on entities and posts makes sure only visible things get added to the query.
+     * @param Builder $query
+     * @return Builder
+     */
+    public function scopePrepareCount(Builder $query): Builder
+    {
+        return $query->where(function ($sub) {
+            return $sub
+                ->where(function ($subEnt) {
+                    return $subEnt
+                        ->entity()
+                        ->has('entity');
+                })
+                ->orWhere(function ($subPost) {
+                    return $subPost
+                        ->post()
+                        ->has('post.entity');
+                });
+        });
+    }
+    /**
+     * @param Builder $query
+     * @return Builder
+     */
+    public function scopeEntity(Builder $query): Builder
+    {
+        return $query->whereNotNull('image_mentions.entity_id');
+    }
+
+    /**
+     * @param Builder $query
+     * @return Builder
+     */
+    public function scopePost(Builder $query): Builder
+    {
+        return $query->whereNotNull('image_mentions.entity_note_id');
+    }
+
+    public function linkToEntity(): string
```

Comment:
Having a Model (part of the db) know about HTML is dirty. I know some older parts of Kanka have that, but we should avoid putting html in models. Please move the html part into the view, thanks :)